hdr/collectors.h

```diff
@@ -0,0 +1,82 @@
+
+#ifndef SQLITE_COLLECTORS_H
+#define SQLITE_COLLECTORS_H
+
+#include <sqlite_modern_cpp.h>
+
+#if __cplusplus > 201402
+#define CPP14_SUPPORTED
+#endif
+
+#ifdef CPP14_SUPPORTED
+#define __COLLECTOR_RETURN_TYPE auto
+#define __COLLECTOR_CTOR_OF_T(T,X)		{X}
+#else
+#define __COLLECTOR_RETURN_TYPE 	std::function<void(Args...)>
+#define __COLLECTOR_CTOR_OF_T(T,X)		T(X)
```

Comment:
Don't these names violate the [standard](https://stackoverflow.com/a/228797/6834058)?